src/assets.rs

```diff
@@ -0,0 +1,72 @@
+use bevy::asset::{HandleId, LoadState};
+use bevy::prelude::*;
+
+use crate::ui::{MONOSPACE_FONT, PROPORTIONAL_FONT};
+use crate::GameState;
+
+pub struct LoadingPlugin;
+
+impl Plugin for LoadingPlugin {
+    fn build(&self, app: &mut App) {
+        app.add_startup_system(|asset_server: Res<AssetServer>| {
+            asset_server.watch_for_changes().unwrap()
+        });
+        app.add_system_set(SystemSet::on_enter(GameState::LoadingAssets).with_system(load_assets))
+            .add_system_set(
+                SystemSet::on_update(GameState::LoadingAssets).with_system(update_assets),
+            );
+    }
+}
+
+pub struct GameAssets {
+    font: Handle<Font>,
+    monospace_font: Handle<Font>,
+}
+
+#[allow(dead_code)]
```

Comment:
Please mark future removal by comment.

Reply:
this "allow dead code" will probably just stay because it is needed to inject the font handles into bevy systems.